src/dso_api/dynamic_api/filterset.py

```
@@ -72,19 +75,60 @@ def filter_to_camel_case(value):
+    for f in model._meta.get_fields():
+        if isinstance(f, models.fields.Field):
+            fields[f.attname] = _get_field_lookups(f)
+        if isinstance(f, ForeignKey):
+            fields[f.attname] = _get_field_lookups(f)  # backwards compat
+
+            # In case of a composite FK, get the loose relations
+            # associated with this ForeignKey object and add dotted syntax.
+            schema_field = get_field_schema(f)
+            prefix = f.attname.removesuffix("_id")
+            try:
+                for s in schema_field.sub_fields:
+                    related_field_name = to_snake_case(s.id)
+                    model_field_name = f"{prefix}_{related_field_name}"
+                    try:
+                        model_field = model._meta.get_field(model_field_name)
+                    except FieldDoesNotExist:
+                        # the field is not part of the compound identifier
+                        # we only support filtering on identifiers for now
+                        continue
+
+                    filter_class = dso_filters.DSOFilterSet.FILTER_DEFAULTS[model_field.__class__][
+                        "filter_class"
+                    ]
+
+                    for lookup in _get_field_lookups(model_field):
+                        sub_filter_name = f"{prefix}.{related_field_name}"
+                        if lookup not in ["contains", "exact"]:
+                            sub_filter_name = f"{sub_filter_name}[{lookup}]"
+
+                        filters[sub_filter_name] = filter_class(
+                            field_name=model_field_name,
+                            lookup_expr=lookup,
+                            label=dso_filters.DSOFilterSet.FILTER_HELP_TEXT.get(
+                                filter_class, lookup
+                            ),
+                        )
+            except ValueError:
+                # The schema field has no sub fields
+                pass
```

In cases where the exception is silenced (eg a single `pass` statement) I prefer the following syntax:

```python
from contextlib import suppres
...
with suppress(ValueError):
   ...
```
This makes it clear, right from the start that we are going to ignore `ValueErrors`. Whereas with a `try` statement the expection is that we are going to catch it and doing something with it, only to read after 16 lines of code we are going to ignore it afterall.

---

Anyway it is just a preference. It's okay to leave it as-is. Just wrote it as not many people are aware of `suppress`' existence.

---

nice, I did not know about this